include/openthread.h

```diff
@@ -868,8 +868,21 @@ typedef void (*otStateChangedCallback)(uint32_t aFlags, void *aContext);
  * @param[in]  aCallback  A pointer to a function that is called with certain configuration or state changes.
  * @param[in]  aContext   A pointer to application-specific context.
  *
+ * @retval kThreadError_None    Added the callback to the list of callbacks.
+ * @retval kThreadError_NoBufs  Could not add the callback due to resource constraints.
+ *
+ */
+ThreadError otSetStateChangedCallback(otInstance *aInstance, otStateChangedCallback aCallback, void *aContext);
```

Comment:
What would you think about not adding the `otRemoveStateChangeCallback` function and instead treat a NULL `aCallback` as the remove? I bring it up for selfish reasons, because I have already implemented a whole API with the current function signatures for Windows.